glean-core/ios/GleanTests/DispatchersTest.swift

```diff
@@ -106,4 +106,76 @@ class DispatchersTest: XCTestCase {
             )
         }
     }
+
+    func testCancelBackgroundTasks() {
+        // Set testing mode to false to allow for background execution
+        Dispatchers.shared.setTestingMode(enabled: false)
+
+        // Set task queueing to true to ensure that we clear out the queued tasks when we cancel
+        Dispatchers.shared.setTaskQueuing(enabled: true)
+
+        // Add a task that should be queued
+        Dispatchers.shared.launch {
+            print("A queued task")
+        }
+
+        // Assert the task was queued
+        XCTAssertEqual(Dispatchers.shared.preInitOperations.count, 1, "Task must be queued")
+
+        // Now cancel the tasks
+        Dispatchers.shared.cancelBackgroundTasks()
+
+        // Assert the task was removed from the queue
+        XCTAssertEqual(Dispatchers.shared.preInitOperations.count, 0, "Task must be removed")
+
+        // Set up our test conditions for normal execution by setting queuing to false
+        Dispatchers.shared.setTaskQueuing(enabled: false)
+
+        // Create some counters to use to determine if the tasks were actually cancelled
```

Comment:
Can you split this into 2 tests? First testing that we clear preInit operations, second that we clear others.

(Maybe even split into preInit/sync/async?)